### PR TITLE
tests/unittests: test black corner case for color_rgb2hsv()

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Tests a (currently undocumented) behavior of `color_rgb2hsv()`, that a
black RGB value (all zero) causes the HSV value to be all zero.

### Testing procedure
Run the unittests for `color` (`make -C tests/unittests tests-color test`). With current master they will fail, but #9939 fixes them. 

### Issues/PRs references
Requires #9939 for the test to succeed.

### CODE REVIEW

File: tests/unittests/tests-color/tests-color.c [hunk 1]
```diff
@@ -73,6 +73,18 @@ static void test_rgb2hex__success(void)
...
+static void test_rgb2hsv__black(void)
+{
+    color_hsv_t hsv;
+    color_rgb_t rgb = { .r = 0x00, .g = 0x00, .b = 0x00 };
+
+    color_rgb2hsv(&rgb, &hsv);
+
+    TEST_ASSERT(0.0f == hsv.h);
+    TEST_ASSERT(0.0f == hsv.s);
+    TEST_ASSERT(0.0f == hsv.v);
```
Comment (gebart) [inline tests/unittests/tests-color/tests-color.c]: You should never compare floats for equality. Use the absolute value and a tolerance instead.

**Comment by pekkanikander** [comment]
Mostly FYI in the case someone is interested.  I converted the HSV data structures partially into fixed point, since I suspected that floats could have caused instability.  The result is in this commit:

https://github.com/AaltoNEPPI/RIOT/commit/73892471300036c4ad874d5c83f69a504d7a4ac8

It is not ready to land, but maybe someone wants to pick it up or have a look at it.  I am not planning to continue with it; if anyone feels like grabbing, please go ahead.

File: tests/unittests/tests-color/tests-color.c (hunk 1 above)
Comment (miri64) [reply tests/unittests/tests-color/tests-color.c]: Better?

Comment (pekkanikander) [reply tests/unittests/tests-color/tests-color.c]: I'd use something like
```
   TEST_ASSERT(hsv.h < EPSILON);
```
where `EPSILON` is a "small enough" float constant.  What is small enough depends on the application and I don't remember the HSV code here, but maybe
```
   const float EPSILON = 1E-10f;
```

Comment (miri64) [reply tests/unittests/tests-color/tests-color.c]: Done.

**Comment by miri64** [comment]
@gebart I adapted the PR according to @pekkanikander's proposal. Can you have another look?

**Comment by miri64** [comment]
@kaspar030 could you maybe have a look at this :-/?

**Comment by miri64** [comment]
@benpicco you want to maybe have a look at this :grin:

**Comment by benpicco** [comment]
It should suffice to test if `hsv.v` is within *epsilon* of 0.

Screenshot at 2019-09-26 22-59-34

`hsv.h = 0` would be red anway.

**Comment by miri64** [comment]
True, but that doesn't test the case this PR was originally introduced for: https://github.com/RIOT-OS/RIOT/pull/9939

**Comment by benpicco** [comment]
Well technically they could be any random value…
`hsv.s = 0` makes sense to stay on the black - white spectrum, but hue should be truly irrelevant.

Maybe someone comes up with a better implementation where it's left uninitialized, that would still be valid.

**Comment by miri64** [comment]
> Well technically they could be any random value…
> `hsv.s = 0` makes sense to stay on the black - white spectrum, but hue should be truly irrelevant.
> 
> Maybe someone comes up with a better implementation where it's left uninitialized, that would still be valid.

so what are you asking?

**Comment by benpicco** [comment]
I'd remove the test for `hsv.h == 0` since it's value is undefined in this case.

**Comment by miri64** [comment]
Done

**Comment by benpicco** [comment]
Looks good, please squash!

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
I believe @gebart's comments were addressed.

**Comment by benpicco** [comment]
tbh, comparing a `float` with `0` to test if it's `0` should always be fine.
